hnn_core/network.py

```diff
@@ -870,17 +870,26 @@ def _instantiate_drives(self, n_trials=1):
                 event_times = list()  # new list for each trial and drive
 
                 # loop over drive 'cells' and create event times for each
-                for this_cell_drive_conn in drive['conn'].values():
-                    for drive_cell_gid in this_cell_drive_conn['src_gids']:
-                        event_times.append(_drive_cell_event_times(
-                            drive['type'], this_cell_drive_conn,
-                            drive['dynamics'], trial_idx=trial_idx,
-                            drive_cell_gid=drive_cell_gid,
-                            seedcore=drive['seedcore'])
-                        )
-                    # only create one event_times list for globals
-                    if not drive['cell_specific']:
-                        break  # loop over drive['conn'].values
+                if drive['cell_specific']:
+                    for this_cell_drive_conn in drive['conn'].values():
+                        for drive_cell_gid in this_cell_drive_conn['src_gids']:
+                            event_times.append(_drive_cell_event_times(
+                                drive['type'], this_cell_drive_conn,
+                                drive['dynamics'], trial_idx=trial_idx,
+                                drive_cell_gid=drive_cell_gid,
+                                seedcore=drive['seedcore'])
+                            )
+                else:
+                    for this_cell_drive_conn in drive['conn'].values():
+                        for drive_cell_gid in this_cell_drive_conn['src_gids']:
+                            event_ts = _drive_cell_event_times(
+                                drive['type'], this_cell_drive_conn,
+                                drive['dynamics'], trial_idx=trial_idx,
+                                drive_cell_gid=drive_cell_gid,
+                                seedcore=drive['seedcore'])
```

Comment:
Since each `this_cell_drive_conn` has the same `seedcore`, they end up getting the same event times. Perhaps there's a cleaner way to only run this once...